Source/i_video.c

```diff
@@ -974,6 +979,62 @@ int cfg_aspectratio; // haleyjd 05/11/09: aspect ratio correction
 // haleyjd 05/11/09: true if called from I_ResetScreen
 static boolean changeres = false;
 
+// [crispy] re-calculate SCREENWIDTH, SCREENHEIGHT, NONWIDEWIDTH and WIDESCREENDELTA
+void I_GetScreenDimensions(void)
+{
+   SDL_DisplayMode mode;
+   int w = 16, h = 10;
+   int ah;
+
+   SCREENWIDTH = ORIGWIDTH;
+   SCREENHEIGHT = ORIGHEIGHT;
+
+   NONWIDEWIDTH = SCREENWIDTH;
+
+   ah = useaspect ? (6 * SCREENHEIGHT / 5) : SCREENHEIGHT;
+
+   if (SDL_GetCurrentDisplayMode(0/*video_display*/, &mode) == 0)
+   {
+      // [crispy] sanity check: really widescreen display?
+      if (mode.w * ah >= mode.h * SCREENWIDTH)
+      {
+         w = mode.w;
+         h = mode.h;
+      }
+   }
+
+   // [crispy] widescreen rendering makes no sense without aspect ratio correction
+   if (widescreen && useaspect)
+   {
+      // switch(crispy->widescreen)
+      // {
+      //     case RATIO_16_10:
+      //         w = 16;
+      //         h = 10;
+      //         break;
+      //     case RATIO_16_9:
+      //         w = 16;
+      //         h = 9;
+      //         break;
+      //     case RATIO_21_9:
+      //         w = 21;
+      //         h = 9;
+      //         break;
+      //     default:
+      //         break;
+      // }
+
+      SCREENWIDTH = w * ah / h;
+      // [crispy] make sure SCREENWIDTH is an integer multiple of 4 ...
+      SCREENWIDTH = (SCREENWIDTH + 3) & (int)~3;
```

Comment:
I have recently changed this to `SCREENWIDTH = (SCREENWIDTH + (crispy->hires ? 0 : 3)) & (int)~3;` in Crispy, please do the same here.